test/protocol/test/textDocument/semanticTokens/full.test.ts

```diff
@@ -0,0 +1,40 @@
+import { expect } from 'chai'
+import { test } from 'mocha'
+import { TestLanguageClient } from '../../../src/TestLanguageClient'
+import { getInitializedClient } from '../../client'
+import { getProjectPath } from '../../helpers'
+import { toUri } from '../../../src/helpers'
+
+let client!: TestLanguageClient
+
+describe('[hardhat] semanticTokens/full', () => {
+  let testPath: string
+
+  before(async () => {
+    client = await getInitializedClient()
+
+    testPath = getProjectPath('hardhat/contracts/semanticTokens/full/SemanticTokens.sol')
+
+    await client.openDocument(testPath)
+  })
+
+  after(async () => {
+    client.closeAllDocuments()
+  })
+
+  test('provides highlighting for types, keywords, functions, numbers and strings', async function () {
+    const semanticTokens = await client.getSemanticTokensFull(toUri(testPath))
+
+    expect(semanticTokens).to.deep.equal({
+      data: [
+        2, 0, 6, 0, 0, 0, 7, 8, 0, 0, 2, 0, 9, 0, 0, 0, 10, 13, 2, 0, 2, 0, 6, 0, 0, 0, 7, 10, 2, 0, 1, 4, 7, 0, 0, 1,
```

Comment:
I suggest dividing this into groups of 5 (one for each token) for readability:

```suggestion
[
    [2, 0, 6, getTokenTypeIndex(SemanticTokenTypes.class), 0],
    [0, 7, 8. getTokenTypeIndex(SemanticTokenTypes.enum), 0],
    ...
].flat()
```

Reply:
I'm afraid this could be too time consuming for what it's worth. My thought process is, once we verify locally that all symbols are highlighted properly (by running the test client), we can dump the LSP response and put it into the assertion. Then if at any point the logic changes, the test will fail. To achieve what you suggest the easiest way I can think of is fabricating cst-like nodes on the test case for increased readability and leveraging SemanticTokensBuilder class for serializing, and then performing the assertion. But I'm not sure it's worth the effort

Reply:
I see. Sounds good. Thanks.